### DISCUSSION TITLE
Allow running headless editor scripts

### DISCUSSION THREAD

**Discussion by JorensM** [godotengine/godot-proposals#8664] [Editor]
### Describe the project you are working on

[Someone on Reddit](https://www.reddit.com/r/godot/s/FlnS5XXc2L) wants to run a headless script from CMD that extends EditorScript. The person faced the problem that the script must extend `SceneTree` or `MainLoop`, but when they do that, the editor specific functions are not available anymore.

### Describe the problem or limitation you are having in your project

Cannot run headless editor script from CMD

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allowing to run headless editor scripts would open opportunity for new ways to utilize headless scripts.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Don't know how it would work as I don't know the inner workings of Godot, sorry

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Not that I know of

### Is there a reason why this should be core and not an add-on in the asset library?

This should be in core because to me it seems like a must-have feature for core.

**Comment by AThousandShips** [comment]
As far as I know the editor can't generally be run in headless mode except for certain stuff like exporting and generation, but those are all specific modes that don't utilize scripts, so what functionality of the eidtor would make sense to use in a headless mode? None of the editor core functionality works with it in that mode, as it all relies on the editor interface, what specific behavior would you use?

You don't mention *why* you need this functionality, please explain this

Since this has no actual ideas for how this should be done I'll go ahead and convert it to a discussion

**Reply by JorensM** [reply]
In the Reddit post I mentioned, the person wants to import assets.

In the [Godot docs](https://docs.godotengine.org/en/stable/tutorials/editor/command_line_tutorial.html), under Running a Script section, it says: 

"This feature is especially useful in large projects, e.g. for batch conversion of assets or custom import/export."

**Comment by KoBeWi** [comment]
It's possible to open the editor in a specific scene, e.g. `godot.exe --editor --headless res://SomeScene.tscn`
Attach a tool script in the root and you are able to perform operations. EditorInterface is a singleton in 4.2, so you are able to access all editor context.
